lib/bolt/config.rb

```diff
@@ -18,17 +18,24 @@ module Bolt
       format: 'human'
     }.freeze
 
-    TRANSPORT_OPTIONS = %i[insecure password run_as sudo_password extensions
-                           key tty tmpdir user connect_timeout cacert
+    TRANSPORT_OPTIONS = %i[host_key_check password run_as sudo_password extensions
+                           ssl key tty tmpdir user connect_timeout cacert
                            token_file orch_task_environment service_url].freeze
 
     TRANSPORT_DEFAULTS = {
       connect_timeout: 10,
       orch_task_environment: 'production',
-      insecure: false,
       tty: false
     }.freeze
 
+    SHH_DEFAULTS = {
```

Comment:
Should be `SSH_DEFAULTS`

Reply:
This should maybe be refactored to something like:

```rb
TRANSPORT_SPECIFIC_DEFAULTS = {
  ssh: {
    host_key_check: true
  },
  winrm: {
    ssl: true
  }
}
```

Then the code below doesn't need to have an explicit switch based on transport.

Reply:
@nicklewis I'm not sure if what I did addresses your concern or not.  I think this could generally use a refactor though.  If my newest change works for you I'll fixup the last commit.